### PR TITLE
busctl: various bugfixes + tweaks

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #34048
Replaces: #34796
Follow-up for: #33961

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: shell-completion/bash/busctl [hunk 1]
```diff
@@ -89,7 +89,7 @@ _busctl() {
                       --allow-interactive-authorization=no --augment-creds=no
                       --watch-bind=yes -j -l --full --xml-interface'
         [ARG]='--address -H --host -M --machine --match --timeout --size --json
-                      --destination --num-matches'
+                      --destination --limit-messages'
```
Comment (yuwata) [inline shell-completion/bash/busctl]: Please add -N.

File: shell-completion/zsh/_busctl [hunk 2]
```diff
@@ -310,5 +310,5 @@ _arguments \
     '--allow-interactive-authorization=[Allow interactive authorization for operation]:boolean:(1 0)' \
     '--timeout=[Maximum time to wait for method call completion and monitoring]:timeout (seconds)' \
     '--augment-creds=[Extend credential data with data read from /proc/$PID]:boolean:(1 0)' \
-    '--num-matches=[Exit after receiving a number of matches while monitoring]:integer' \
+    '--limit-messages=[Exit after receiving a number of matches while monitoring]:integer' \
```
Comment (yuwata) [inline shell-completion/zsh/_busctl]: Also here.

**Review by yuwata — APPROVED** [review]

File: src/busctl/busctl.c [hunk 3]
```diff
@@ -2503,7 +2507,8 @@ static int help(void) {
                "     --watch-bind=BOOL     Wait for bus AF_UNIX socket to be bound in the file\n"
                "                           system\n"
                "     --destination=SERVICE Destination service of a signal\n"
-               "     --num-matches=NUMBER  Exit after receiving a number of matches while\n"
+               "  -N --limit-messages=NUMBER\n"
+               "                           Exit after receiving a number of matches while\n"
```
Comment (dtardon) [inline src/busctl/busctl.c:2511]: "matches" -> "messages"

Comment (dtardon) [reply src/busctl/busctl.c:2511]: Also, maybe rephrase this as "Stop monitoring after receiving a specified number of messages"?

File: shell-completion/zsh/_busctl (hunk 2 above)
Comment (dtardon) [inline shell-completion/zsh/_busctl]: "matches" -> "messages"

Comment (poettering) [reply shell-completion/zsh/_busctl]: uh, that's a big ask, i have no frickin clue about this language, and it's *weird* with it's `'` use and everything. I cannot reasonably test this, I am pretty sure we should not ask for everyone to learn the syntax of some niche shell to submit a patch like this "zsh" thing.

Comment (bluca) [reply shell-completion/zsh/_busctl]: yeah stuff like this can be fixed later, no need to be an RC blocker
